src/ConnectionTrait.php

```diff
@@ -205,24 +209,17 @@ public function prepare($sql)
      */
     protected function prepareWrapped($sql)
     {
-        return new Statement($sql, $this);
-    }
+        $stmt = new Statement($sql, $this);
+        $stmt->setFetchMode($this->defaultFetchMode);
```

Comment:
Isn't this a BC?

Reply:
No it isn't, because before these changes our Statement decorated the original one created calling the Connection parent method and the default fetch mode was set there.